neurom/core/_neuron.py

```diff
@@ -28,16 +28,176 @@
+            return self.morphio_section.append_section(section.morphio_section)
+        return self.morphio_section.append_section(section)
+
+    def is_forking_point(self):
+        """Is tree a forking point?"""
+        return len(self.children) > 1
+
+    def is_bifurcation_point(self):
+        """Is tree a bifurcation point?"""
+        return len(self.children) == 2
+
+    def is_leaf(self):
+        """Is tree a leaf?"""
+        return len(self.children) == 0
+
+    def is_root(self):
+        """Is tree the root node?"""
+        return self.parent is None
+
+    def ipreorder(self):
+        """Depth-first pre-order iteration of tree nodes."""
+        children = deque((self, ))
+        while children:
+            cur_node = children.pop()
+            children.extend(reversed(cur_node.children))
+            yield cur_node
+
+    def ipostorder(self):
+        """Depth-first post-order iteration of tree nodes."""
+        children = [self, ]
+        seen = set()
+        while children:
+            cur_node = children[-1]
+            if cur_node not in seen:
+                seen.add(cur_node)
+                children.extend(reversed(cur_node.children))
+            else:
+                children.pop()
+                yield cur_node
+
+    def iupstream(self):
+        """Iterate from a tree node to the root nodes."""
+        t = self
+        while t is not None:
+            yield t
+            t = t.parent
+
+    def ileaf(self):
+        """Iterator to all leaves of a tree."""
+        return filter(Section.is_leaf, self.ipreorder())
+
+    def iforking_point(self, iter_mode=ipreorder):
+        """Iterator to forking points.
+
+        Args:
+            iter_mode: iteration mode. Default: ipreorder.
+        """
+        return filter(Section.is_forking_point, iter_mode(self))
+
+    def ibifurcation_point(self, iter_mode=ipreorder):
+        """Iterator to bifurcation points.
+
+        Args:
+            iter_mode: iteration mode. Default: ipreorder.
+        """
+        return filter(Section.is_bifurcation_point, iter_mode(self))
+
+    def __eq__(self, other):
+        """Equal when its morphio section is equal."""
+        return self.morphio_section == other.morphio_section
+
+    def __hash__(self):
+        """Hash of its id."""
+        return hash(self.id)
+
+    def __nonzero__(self):
+        """If has children."""
+        return bool(self.children)
+
+    __bool__ = __nonzero__
+
+    @property
+    def points(self):
+        """Returns the section list of points the NeuroM way (points + radius)."""
+        return np.concatenate((self.morphio_section.points,
+                               self.morphio_section.diameters[:, np.newaxis] / 2.),
+                              axis=1)
+
+    @points.setter
+    def points(self, value):
+        """Set the points."""
+        self.morphio_section.points = np.copy(value[:, COLS.XYZ])
+        self.morphio_section.diameters = np.copy(value[:, COLS.R]) * 2
```

Comment:
this is dangerous, right? If I do:
```
section.points[10] = [1, 2, 3, 4]
```
then nothing is updated, right? I had this issue with morphio that one needs to give the entire set of points to be able to update them, due to underlying c code. I wonder if here we can avoid this?

Reply:
> then nothing is updated, right?

Right. Not much I can do here. All points must be updated all at once.

Reply:
I see, ok, thanks!